embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java

```diff
@@ -218,6 +220,10 @@ public ConfigDiff transaction(ConfigSource config,
             // TODO incremental_columns is not set => get primary key
             schema = setupTask(con, task);
         } catch (SQLException ex) {
+            final Throwable cause = ex.getCause();
+            if (cause instanceof SocketTimeoutException || cause instanceof UnknownHostException) {
```

Comment:
It is understandable `UnknownHostException` to consider as `ConfigException`. But `SocketTimeoutException` does not sound like a `ConfigException` actually.

(Even when we consider a retry based on a type of Exception (sorry, it's our internal use-case, @hito4t), socket timeout is considered to recover by some retries?)

Reply:
When i am testing on local with invalid host, i also get both these exceptions with invalid `host`
- `UnknownHostException` when internet issue,  unable to resolve internet addresses
- `SocketTimeoutException` when server issue, fail to response

imo, those exceptions have chance to recover by some retries, we ignore only when start `transaction`

Reply:
The problem here is NOT that you received those exceptions when testing on your local environments. The problem here is that real tentative network unstability is also (unintentionally or unexpectedly) affected by this.

Reply:
removed SocketTimeoutException